src/App.tsx

```diff
@@ -1,28 +1,109 @@
+import { todoActions } from './store/сurrentTodo';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(Filter.ALL);
+  const [savedTodo, setSavedTodo] = useState<Todo>();
+
+  const dispatch = useDispatch();
+  const isTodoLoading = useSelector(selectors.isLoading);
+  const modalIsOpened = useSelector(selectors.getTodo);
+
+  useEffect(() => {
+    getTodos().then(todoList => {
+      setTodos(todoList);
+      setVisibleTodos(todoList);
+      dispatch(loadActions.startLoading());
+    });
+  }, []);
+
+  const searchByWord = (title: string) => {
+    const params = query.toLowerCase();
+
+    return title.toLocaleLowerCase().includes(params);
+  };
+
+  useEffect(() => {
+    switch (filter) {
+      case Filter.ALL:
+        setVisibleTodos(todos.filter(todo => searchByWord(todo.title)));
+        break;
 
-export const App = () => {
-  // `useSelector` connects our component to the Redux store
-  // and rerenders it after every dispatched action
-  const loading = useSelector(selectors.isLoading);
+      case Filter.ACTIVE:
+        setVisibleTodos(todos.filter(todo => !todo.completed && searchByWord(todo.title)));
+        break;
 
-  // we do not call a selector with (), just pass a link to it
-  const message = useSelector(selectors.getMessage) || 'Ready!';
+      case Filter.COMPLITED:
+        setVisibleTodos(todos.filter(todo => todo.completed && searchByWord(todo.title)));
+        break;
+      default:
+        break;
+    }
+  }, [query, filter]);
+
+  const todoOpener = (todo: Todo) => {
+    setSavedTodo(todo);
+    dispatch(todoActions.todoSelect(todo));
+  };
+
+  const getQuery = (userQuery: string) => {
+    setQuery(userQuery);
+  };
+
+  const selectedFilter = (userFilter: string) => {
+    setFilter(userFilter);
+  };
 
   return (
-    <div className="App">
-      <h1>Redux list of todos</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
-
-      {/* these buttons are used only for the demo */}
-      <Start title="Start loading" />
-      <Finish title="Succeed" message="Loaded successfully!" />
-      <Finish title="Fail" message="Error occurred." />
-    </div>
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter
+                getQuery={getQuery}
+                selectedFilter={selectedFilter}
+              />
+            </div>
+
+            <div className="block">
+              {!isTodoLoading
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    todoOpener={todoOpener}
+                  />
+                )}
```

Comment:
Wrong logic. You should render loader when variable `isTodoLoading` equals `true`,